src/main/java/io/logz/logback/LogzioSender.java

```diff
@@ -125,6 +127,26 @@ public LogzioSender(String logzioToken, String logzioType, int drainTimeout, int
         debug("Created new LogzioSender class");
     }
 
+    public static LogzioSender getOrCreateSenderByType(String logzioToken, String logzioType, int drainTimeout, int fsPercentThreshold, String bufferDir,
+                                                       String logzioUrl, int socketTimeout, int connectTimeout, boolean debug,
+                                                       LogzioLogbackAppender.StatusReporter reporter, ScheduledExecutorService tasksExecutor,
+                                                       boolean addHostname, String additionalFields, int gcPersistedQueueFilesIntervalSeconds) {
+
+        LogzioSender logzioSenderInstance = logzioSenderInstances.get(logzioType);
```

Comment:
This all section is not thread safe.

Also - I don't understand the motivation here to keep a singleton sender per type - why and why type?